activation/activation.go

```diff
@@ -720,7 +724,7 @@ func (b *Builder) GetPositioningAtxInfo() (types.ATXID, types.LayerID, error) {
 	if err != nil {
 		if errors.Is(err, sql.ErrNotFound) {
 			b.log.With().Info("using golden atx as positioning atx", b.goldenATXID)
-			return b.goldenATXID, types.LayerID{}, nil
+			return b.goldenATXID, types.NewLayerID(0), nil
```

Comment:
what's the diff btwn types.LayerID{} and types.NewLayerID(0)?

Reply:
Nothing, it is the same. The difference is here it actually means layer 0, while below it means n/a.

Reply:
i find this confusing if the end result is the same. differentiating here means you should differentiate it everywhere else.

Reply:
There is no way to return "no layer" since 0 is a valid layer and `-1` would panic. I can make this type a pointer and return `nil` instead below.

Imo it was confusing before: why does it return the same here as below in the case of an error?

Reply:
i agree its confusing before. but only changing it here make it more confusing. the code has many other place that does this. that was my point  : P